### ISSUE TITLE
Encoding error with non-UTF-8 PDFs

### ISSUE THREAD

**Issue by PierreMesure** [metebalci/pdftitle#49]
Hi,

I'm getting errors with PDFs encoded with latin-1. Here's an [example](https://web.archive.org/web/20170428083007/https://www.aklagare.se/globalassets/dokument/rapporter/ovriga-rapporter/2015_rapport-aldersbestamning-unga-lagovertradare.pdf.pdf).

The problem occurs at [this line](https://github.com/metebalci/pdftitle/blob/daefa5a23147352d774c45b3089026a1af8db66a/pdftitle/metadata.py#L35) because the byte string isn't encoded with UTF-8. If I replace with `title.decode('iso-8859-1')`, it works flawlessly.

I think a solution would be to extract the info encoding using pdfminer but I couldn't find how. Another possibility is using chardet or testing for several encodings and catching the exceptions.

**Comment by user202729** [comment]
Same issue with https://arxiv.org/pdf/2004.07558v3 . Looks like for this one the title is in UTF16 though.

`pdfinfo` can recover the title successfully however. It might be worth looking into pdfinfo implementation to see how it detects the encoding.

**Comment by metebalci** [comment]
Good catch and thanks for the examples to be used as test cases. So as you found out, it was assumed to be always utf-8, whereas it could be PdfDocEncoding, UTF-16BE or UTF-8. I change the implementation to handle all three and it works correct for these examples.
